front/tests/chat/e2e_eval.ts

```diff
@@ -0,0 +1,173 @@
+  const auth = await Authenticator.internalBuilderForWorkspace(workspacesId);
+  const owner = auth.workspace();
+  if (!owner) {
+    throw new Error("Invalid workspace");
+  }
+
+  async function computeChatAnswer(input: ChatEvalInputType) {
+    /* Call the chat lib */
+    const chatRes = await newChat(
+      auth,
+      {
+        userMessage: input.question,
+        dataSources: null,
+        filter: null,
+        timeZone: "Europe/Paris",
+      },
+      false
+    );
+    let answer = undefined;
+    for await (const event of chatRes) {
+      if (
+        event.type === "chat_message_create" &&
+        event.message.role === "assistant"
+      ) {
+        answer = event.message.message;
+      } else if (event.type === "chat_session_create") {
+        if (verbose) console.log(event.session);
+      }
+    }
+
+    return answer ? answer : "There was an error. Please check with the team";
+  }
+
+  /* Load the JSON test data from data/chat-eval-inputs.jsonl */
+  let evalData = JSON.parse(
+    fs.readFileSync(path.join(__dirname, "data/chat-eval-inputs.json"), "utf8")
+  );
+  if (numberOfQuestions)
+    evalData = evalData.slice(0, numberOfQuestions) as ChatEvalInputType[];
+
+  /* Compute Chat's answers for each input */
+  const dataWithAnswers = await Promise.all(
+    evalData.map(async (input: ChatEvalInputType) => {
+      if (verbose) console.log(`Computing answer for ${input.question}`);
+      return {
+        question: input.question,
+        rules: input.rules,
+        answer: await computeChatAnswer(input),
+      };
+    })
+  );
+
+  /* run the chat eval app on dataWithAnswers */
+  const config = cloneBaseConfig(
+    DustProdActionRegistry["chat-message-e2e-eval"].config
+  );
+  const evalResult = (await runAction(
+    auth,
+    "chat-message-e2e-eval",
+    config,
+    dataWithAnswers
+  )) as Ok<RunType>;
+
+  if (!evalResult.value.results) {
+    throw new Error("No results found");
+  }
+  /* Format the results */
+  return evalResult.value.results.map((r, i) => {
+    return {
+      question: dataWithAnswers[i].question,
+      answer: dataWithAnswers[i].answer,
+      evalResult: (r[0].value as { result: object; rule: string }[]).map(
+        (v) => {
+          return {
+            ...v.result,
+            rule: v.rule,
+          } as EvalResultType;
+        }
+      ),
+    };
+  });
+}
+
+async function main() {
+  const numberOfQuestions = process.argv.includes("--n")
+    ? parseInt(process.argv[process.argv.indexOf("--n") + 1])
+    : undefined;
+  const verbose = process.argv.includes("--verbose");
+  const chatEvalResults = await chatEval(numberOfQuestions, verbose);
+
+  /* count the number of successes */
+  const successfulQuestion = (r: EvalResultType[]) => {
+    return r.reduce((acc, r) => {
+      return acc && r.rule_respected === "YES";
+    }, true);
+  };
+
+  const successes = chatEvalResults.reduce((acc, r) => {
```

Comment:
NIT: rename to `totalSuccesses ` or `numberOfSuccesses` so that we know instantly it stores the total number?